include/seqan3/search/search.hpp

```diff
@@ -165,6 +168,26 @@ inline auto search(queries_t && queries,
 }
 
 //!\cond DEV
+// Convert query sequence if it does not match the alphabet type of the index.
+//!\overload
+template <fm_index_specialisation index_t,
+          std::ranges::forward_range queries_t,
+          typename configuration_t = decltype(search_cfg::default_configuration)>
+    requires std::ranges::forward_range<std::ranges::range_reference_t<queries_t>> &&
+             (!std::same_as<range_innermost_value_t<queries_t>, typename index_t::alphabet_type>)
+inline auto search(queries_t && queries,
+                   index_t const & index,
+                   configuration_t const & cfg = search_cfg::default_configuration)
+{
+    static_assert(std::convertible_to<range_innermost_value_t<queries_t>, typename index_t::alphabet_type>,
+                  "The alphabet of the text collection must be convertible to the alphabet of the index.");
+
+    if constexpr (range_dimension_v<queries_t> == 2u)
+        return search(queries | views::deep{views::convert<typename index_t::alphabet_type>}, index, cfg);
```

Comment:
Why do we need the deep function here?

Reply:
Because we can pass multiple queries, i.e. a vector of queries: `{"ACGT"_dna4, "GGTA"_dna4}`
If it's a range of queries, we need to convert each query (deep).
If it's a single query, we just convert the query.